### ISSUE TITLE
Potentially false negative for constructor properties for  `spacing-between-declarations-with-annotations` rule

### ISSUE THREAD

**Issue by Kantis** [pinterest/ktlint#1579]
## Expected Behavior
The following code only reports an error on the commented line:
```kotlin
@Serializable
data class EmploymentAgreement(
   @Contextual
   val firstEmploymentDate: LocalDate,
   val employmentLevel: Double,
   @Contextual
   val activeFrom: LocalDate,
   @Contextual // Why no issue here?
   val activeTo: LocalDate?,
   val vacationDays: String,
   @Contextual
   val employmentTerminationDate: LocalDate?,
   val employmentType: String,
) {
   @Serializable
   enum class EmploymentType {
      @SerialName("Visstidsanställning")
      Hourly,
      @SerialName("Tillsvidareanställning") // Only this line reports an issue with Ktlint
      Full
   }
}
```

I would expect it to be formatted as:

```kotlin
@Serializable
data class EmploymentAgreement(

   @Contextual
   val firstEmploymentDate: LocalDate,
   val employmentLevel: Double,

   @Contextual
   val activeFrom: LocalDate,

   @Contextual // Why no issue here?
   val activeTo: LocalDate?,
   val vacationDays: String,

   @Contextual
   val employmentTerminationDate: LocalDate?,
   val employmentType: String,
) {
   @Serializable
   enum class EmploymentType {
      @SerialName("Visstidsanställning")
      Hourly,

      @SerialName("Tillsvidareanställning") // Only this line reports an issue with Ktlint
      Full
   }
}
```

## Observed Behavior
The code is formatted as:

```kotlin
@Serializable
data class EmploymentAgreement(
   @Contextual
   val firstEmploymentDate: LocalDate,
   val employmentLevel: Double,
   @Contextual
   val activeFrom: LocalDate,
   @Contextual // Why no issue here?
   val activeTo: LocalDate?,
   val vacationDays: String,
   @Contextual
   val employmentTerminationDate: LocalDate?,
   val employmentType: String,
) {
   @Serializable
   enum class EmploymentType {
      @SerialName("Visstidsanställning")
      Hourly,

      @SerialName("Tillsvidareanställning") // Only this line reports an issue with Ktlint
      Full
   }
}
```

## Your Environment

* Version of ktlint used: 0.46.1
* Using CLI
* Operating System and version: macOS 12.5

**Comment by paul-dingemans** [comment]
I agree with you that this feels inconsistent. There is however a subtle difference between `activeTo` and `Full`. The fields in the data class are not considered to be declarations. The values in the enum values are declarations. 

The [Kotlin Coding conventions](https://kotlinlang.org/docs/coding-conventions.html#file-annotations) do not require a blank line between a declaration and the annotation of the next declaration. So adding that as a rule to ktlint would not be acceptable.

I think that this rule has been added in order to be consistent with the default IntelliJ IDEA formatter which inserts this blank line between annotated enum values.

**Comment by Kantis** [comment]
Thanks for checking @paul-dingemans. Feel free to close the issue if there's no improvement that can be made in line with Ktlint's philosophy 🙂

**Comment by blommish** [comment]
@paul-dingemans I see this introduced a new empty line. Where we previously had tab, is now changed to empty line

```java
    @Annotation
    private val ...
<tab>
   @Annotation
    private val ...
```

is changed to
```java
    @Annotation
    private val ...
<empty>
   @Annotation
    private val ...
```

**Comment by paul-dingemans** [comment]
I would say that is expected behavior. Lines should not contain any indentation when nothing else is on that line.
